src/couch/src/couch_server.erl

```diff
@@ -386,6 +410,20 @@ open_async(Server, From, DbName, {Module, Filepath}, Options) ->
     true = ets:insert(couch_dbs_pid_to_name, {Opener, DbName}),
     db_opened(Server, Options).
 
+open_async_int(Server, DbName, Options) ->
+    DbNameList = binary_to_list(DbName),
+    case check_dbname(DbNameList) of
```

Comment:
Minor nit: technically we don't have to pass in a list as it will eventually be translated back into a binary.

https://github.com/apache/couchdb/blob/master/src/couch/src/couch_db.erl#L1775

Reply:
I had contemplated that but I figured a second pass just in case something somewhere somehow managed to bypass the other checks we'd still catch it. I just considered the first pass in open/create to avoid the LRU ejection issue you found.